### ISSUE TITLE
BUG: Kaiser accepts non-integer size

### ISSUE THREAD

**Issue by endolith** [scipy/scipy#6390]
and outputs nans:

```
kaiser(1845.5, 14)
Out[55]: 
array([  7.72686684e-06,   8.56954025e-06,   9.45636923e-06, ...,
         9.00734538e-06,   8.14277352e-06,              nan])
```

should probably raise an exception?  should check if other window functions do the same

**Comment by Eric89GXL** [comment]
I agree it should be an exception if it isn

**Comment by Eric89GXL** [comment]
...t an int-like > 0

**Comment by endolith** [comment]
@Eric89GXL You can edit comments, you know. :D

Currently it explicitly outputs an empty array for N <= 0.  

Probably empty output for N = 0 is expected in some degenerate cases?  But I think an error for negative N would be ok to add.

Otherwise I mostly fixed this in https://github.com/scipy/scipy/compare/master...endolith:window_int

Should add tests though

**Comment by endolith** [comment]
Oh wait I'm dumb, you can't do an early return from within the helper function.  What's the most elegant way to do this?

**Comment by Eric89GXL** [comment]
I like that this makes it more DRY, but you're right to get the early return you'll have to add something like:
```
if isinstance(M, np.ndarray):
    return M
```
after the `_check_*` calls.

**Comment by endolith** [comment]
Yeah I don't see any cleaner way to do it.   http://stackoverflow.com/q/38517374/125507

**Comment by larsoner** [comment]
Fixed in `master` (probably from #6407), now gives:
```
ValueError: Window length M must be a non-negative integer
```
